### DISCUSSION TITLE
Validation Prog ression Bar

### DISCUSSION THREAD

**Discussion by Polarisjame** [Lightning-AI/pytorch-lightning#15163] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
![image](https://user-images.githubusercontent.com/108406757/196378785-a506348b-7579-4848-9e71-ffd45e1e88a6.png)
When I run the Validation step,it seems that my val_loss and val_acc don't show on the prog_bar and it line feeds automaticly ,I don't Know how to deal with it,here is my model code:
```python
    def validation_step(self, batch, batch_idx):
        data_x, data_y = batch
        outputs = self(data_x)
        train_pred = argmax(outputs, dim=1)
        train_acc = train_pred == data_y
        loss = self.loss(outputs, data_y)
        acurate = sum(train_acc) / len(train_acc)
        metrics = {'loss': loss, 'acc': acurate}
        self.log('acc', acurate, on_step=True, on_epoch=False, prog_bar=True)
        return train_acc

    def validation_epoch_end(self, validation_step_outputs):
        # print(validation_step_outputs)
        val_out = [num for elem in validation_step_outputs for num in elem]
        acc = sum(val_out) / len(val_out)
        del val_out
        self.log('acc', acc, on_step=False, on_epoch=True, prog_bar=True)
```
and my trainer code:
```python
train_set = data.train_dataloader()
    val_set = data.val_dataloader()
    trainer = Trainer.from_argparse_args(args, max_epochs=args.epoches, accelerator='gpu', devices=-1,
                                         auto_scale_batch_size=args.auto_batch, auto_lr_find=args.auto_lr, precision=16,
                                         callbacks=[checkpoint_callback], limit_train_batches=10, limit_val_batches=5)
    trainer.fit(model, train_set, val_set)
```

**Comment by akihironitta** [comment]
@Polarisjame I'm not sure if this applies to your case, but I've heard it doesn't work properly in some terminal emulators. To investigate your issue, I would try running a minimal script like [this one](https://github.com/Lightning-AI/lightning/tree/b340859df9d9cdc8f55f4557566754b25971888d/examples/pl_bug_report) first to see whether it's caused by your implementation or not.

Another option is to switch the default progress bar to `RichProgressBar`.

**Reply by Polarisjame** [reply]
Thanks for your reply, I tried to run my code on cmd and it works successfully, but it can't fit on pycharm terminal.I think that's the problem.

**Reply by akihironitta** [reply]
I'm not a PyCharm user, so I don't know much, but you might want to have a look at this:
> PyCharm users will need to enable “emulate terminal” in output console option in run/debug configuration to see styled output. Reference: https://rich.readthedocs.io/en/latest/introduction.html#requirements

https://pytorch-lightning.readthedocs.io/en/1.7.7/api/pytorch_lightning.callbacks.RichProgressBar.html#pytorch_lightning.callbacks.RichProgressBar

**Reply by Polarisjame** [reply]
Thanks for your advice, switch the default progress bar to RichProgressBar does work with PyCharm.

**Comment by binodil** [comment]
I had similar issue in jupyter notebook. Solved using RichProgressBar! Thanks for this discussion:)
